internal/ansible/ansible_manager.go

```diff
@@ -0,0 +1,248 @@
+package ansible
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/apenella/go-ansible/pkg/execute"
+	"github.com/apenella/go-ansible/pkg/options"
+	"github.com/apenella/go-ansible/pkg/playbook"
+	ansibleResults "github.com/apenella/go-ansible/pkg/stdoutcallback/results"
+	"github.com/project-flotta/flotta-device-worker/internal/ansible/model/message"
+
+	"github.com/project-flotta/flotta-device-worker/internal/ansible/dispatcher"
+	pb "github.com/redhatinsights/yggdrasil/protocol"
+)
+
+// AnsibleManager handle ansible playbook execution
+type AnsibleManager struct {
+	dispatcherClient pb.DispatcherClient
+	// defined how to connect to hosts
+	ansiblePlaybookConnectionOptions *options.AnsibleConnectionOptions
+	// defined which should be the ansible-playbook execution behavior and where to find execution configuration.
+	ansiblePlaybookOptions *playbook.AnsiblePlaybookOptions
+	playbook               *playbook.AnsiblePlaybookCmd
+	ansibleDispatcher      *dispatcher.AnsibleDispatcher
+}
+
+type RequiredFields struct {
+	crcDispatcherCorrelationID, returnURL string
+	responseInterval                      uint64
+}
+
+const (
+
+	//Required field names
+	crcDispatcherAttribute    = "crc_dispatcher_correlation_id"
+	returnURLAttribute        = "return_url"
+	responseIntervalAttribute = "response_interval"
+	defaultResponseInterval   = 300
+
+	// Failure types
+	NotInstalled = "ANSIBLE_NOT_INSTALLED"
+	UndefError   = "UNDEFINED_ERROR"
+)
+
+var (
+	buff, buffErr *bytes.Buffer
+
+	playbookExecutor *execute.DefaultExecute
+	playbookCmd      *playbook.AnsiblePlaybookCmd
+	playbookYamlFile = ""
```

Comment:
iiuc, we should be able to run several playbook in parallel.
meaning `HandlePlaybook` can be run concurrently.
If this is the case, we can't rely on global variables.

Reply:
Yes, you're right. I'll fix this.